ethereum/contracts/TablelandRigs.sol

```diff
@@ -49,13 +55,41 @@ contract TablelandRigs is
     // URI for contract info.
     string private _contractInfoURI;
 
+    // TODO Remove the following, but for context, the table schema:
+    /**
+        create table rig_pilot_sessions(
+            id integer primary key, 
+            rig_id integer not null, 
+            owner text not null,
+            pilot_contract text, 
+            pilot_id integer, 
+            start_time integer not null, 
+            end_time integer
+        )
+     */
+
+    // Table prefix for the Rigs pilot sessions table.
+    string private constant RIG_PILOT_SESSIONS_PREFIX = "rig_pilot_sessions";
+
+    // Table ID for the Rigs pilot sessions table.
+    // TODO do we want to have a method for updating this value?
+    uint256 private _rigPilotSessionsTableId;
+
+    // Tracks the Rig `tokenId` to its current `RigPilot`.
+    mapping(uint256 => RigPilot) internal _pilots;
+
+    // Track piloted vs. parked Rigs as a bitlist.
+    // TODO this was implemented for gas reduction purposes while iterating/checking for duplicates in `_pilots`
```

Comment:
Calling this out in case there are better ways to handle this optimization; `_rigsStatus` is a bitlist that tracks active Rigs. It's used for iteration purposes instead of accessing a `RigPilot.startTime` in the `_pilots` mapping, which also indicates if a Rig is in flight vs. parked.

It's still a gas guzzler down in `pilotRig` since it has to check every Rig for pilot duplicates, but doing this dropped around 5M gas (7.5M down to 2.5M)

Reply:
Realizing this needs some significant overhaul...gonna think about it a bit more, but current prices mean $51 USD to park a Rig since we're on L1. Gonna check out that diamond storage impl vs. struct.

Reply:
i left a suggestion below about using two maps